projects/dotcms-field-elements/src/components/dot-binary-file/dot-binary-file.tsx

```diff
@@ -0,0 +1,286 @@
+ *
+ * @export
+ * @class DotBinaryFile
+ */
+@Component({
+    tag: 'dot-binary-file',
+    styleUrl: 'dot-binary-file.scss'
+})
+export class DotBinaryFileComponent {
+    @Element() el: HTMLElement;
+
+    /** Value specifies the value of the <input> element */
+    @Prop({ mutable: true })
+    value = null;
+
+    /** Name that will be used as ID */
+    @Prop() name = '';
+
+    /** (optional) Text to be rendered next to input field */
+    @Prop({ reflectToAttr: true })
+    label = '';
+
+    /** (optional) Placeholder specifies a short hint that describes the expected value of the input field */
+    @Prop({ reflectToAttr: true })
+    placeholder = 'Attach files by dragging & dropping, selecting or pasting them.';
+
+    /** (optional) Hint text that suggest a clue of the field */
+    @Prop({ reflectToAttr: true })
+    hint = '';
+
+    /** (optional) Determine if it is mandatory */
+    @Prop({ reflectToAttr: true })
+    required = false;
+
+    /** (optional) Text that be shown when required is set and condition not met */
+    @Prop() requiredMessage = 'This field is required';
+
+    /** (optional) Text that be shown when the Regular Expression condition not met */
+    @Prop() validationMessage = "The field doesn't comply with the specified format";
+
+    @Prop() URLValidationMessage = 'The specified URL is not valid';
+
+    /** (optional) Disables field's interaction */
+    @Prop({ reflectToAttr: true })
+    disabled = false;
+
+    /** (optional) Describes a type of file that may be selected by the user, separated by comma  eg: .pdf,.jpg  */
+    @Prop({ reflectToAttr: true })
+    accept = '';
+
+    /** (optional) Text that be shown in the browse file button */
+    @Prop({ reflectToAttr: true })
+    buttonLabel = 'Browse';
+
+    @State() status: DotFieldStatus;
+
+    @Event() valueChange: EventEmitter<DotFieldValueEvent>;
+    @Event() statusChange: EventEmitter<DotFieldStatusEvent>;
+
+    private allowedFileTypes = [];
+    private binaryTextValue = '';
+    private errorMessage = '';
+    private binaryTextField: DotBinaryTextField;
+
+    /**
+     * Reset properties of the field, clear value and emit events.
+     */
+    @Method()
+    reset(): void {
+        this.value = '';
+        this.binaryTextField.reset();
+        this.status = getOriginalStatus(this.isValid());
+        this.emitStatusChange();
+        this.emitValueChange();
+    }
+
+    componentWillLoad(): void {
+        this.validateProps();
+        this.status = getOriginalStatus(this.isValid());
+        this.emitStatusChange();
+    }
+
+    componentDidLoad(): void {
+        this.binaryTextField = this.el.querySelector('dot-binary-text-field');
+    }
+
+    @Watch('accept')
+    optionsWatch(): void {
+        const validTypes = checkProp<DotBinaryFileComponent, string>(this, 'accept');
+        this.allowedFileTypes = !!validTypes ? validTypes.split(',') : [];
+    }
+
+    @Listen('_valueChange')
```

Comment:
Do you know what's the difference using `@Listen` instead of just `<dot-binary-text-field onValueChange={this.binaryTextValueChange.bind(this)}`>?

I think that the listen is listening for ALL the `_valueChange` that happen in any component and that's not what we want for this, we should listen for only the value change of the `dot-binary-text-field`.

Also, using `_valueChange` in internal components is confusing, here we can use any event name because is just for internal use.

Reply:
think there is not difference with `onValueChange `, `@Listen` is just the stencil approach, and by default only hear for host events, https://stenciljs.com/docs/events#target.